### PR TITLE
Added lease extension to java sdk and fixed tests

### PR DESCRIPTION
**Pull request by ystxn**
Pull Request type
----
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] WHOSUSING.md
- [ ] Other (please describe):

Changes in this PR
----

Issue #306
- Add lease extension flag
- Fix tests

### CODE REVIEW

File: conductor-clients/java/conductor-java-sdk/conductor-client/build.gradle [hunk 1]
```diff
@@ -36,6 +35,7 @@ java {
 
 test {
     useJUnitPlatform()
+    maxParallelForks = 1
```
Comment (jmigueprieto) [inline conductor-clients/java/conductor-java-sdk/conductor-client/build.gradle:38]: @ystxn why is this required?

This might significantly slow down tests, since there will be no parallel test execution.

Comment (jmigueprieto) [reply conductor-clients/java/conductor-java-sdk/conductor-client/build.gradle:38]: I see now that tests weren't running because they were using JUnit4 annotations.

**Review by jmigueprieto — APPROVED** [review]
LGTM, let's get this merged and then fix the `maxParallelForks = 1`

File: conductor-clients/java/conductor-java-sdk/conductor-client/build.gradle (hunk 1 above)
Comment (ystxn) [reply conductor-clients/java/conductor-java-sdk/conductor-client/build.gradle:38]: this was to fix the non-determinism in this test. it strangely kept passing on my local but kept failing in github's build.
https://github.com/conductor-oss/conductor/actions/runs/12478325215/job/34825744258#step:4:72
